### PR TITLE
feat: Support open all destination port by an annotation

### PR DESCRIPTION
**Pull request by nilo19**
Introduce a new annotation `service.beta.kubernetes.io/azure-destination-port-all` to open all ports instead of service ports in a security rule.

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Introduce a new annotation `service.beta.kubernetes.io/azure-destination-ports-all` to allow all ports instead of service ports in a security rule. This can be useful as some service tags require to open all ports.

feat: Support open all destination port by an annotation

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
feat: Support open all destination port by an annotation
Introduce a new annotation `service.beta.kubernetes.io/azure-destination-ports-all` to allow all ports instead of service ports in a security rule. This can be useful as some service tags require to open all ports.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by zarvd** [comment]
I suspect it won’t clean up the NSG rules when the service is deleted. Could you add an E2E test to check that?

Also, which service tags require all ports to be open? This requirement seems counterintuitive.

**Comment by nilo19** [comment]
> I suspect it won’t clean up the NSG rules when the service is deleted. Could you add an E2E test to check that?
> 
> Also, which service tags require all ports to be open? This requirement seems counterintuitive.

Checked manually. This is not a commonly used annotation so I don't add an e2e test. I was told GenevaActionsNonProd needs this.

File: pkg/provider/loadbalancer/securitygroup/securityrule.go [hunk 1]
```diff
@@ -132,6 +133,10 @@ func ListDestinationPortRanges(r *network.SecurityRule) ([]int32, error) {
 	for _, v := range values {
 		p, err := strconv.ParseInt(v, 10, 32)
 		if err != nil {
+			if v == consts.PortRangeAll {
+				rv = append(rv, consts.PortRangeAllIndicator)
```
Comment (nilo19) [inline pkg/provider/loadbalancer/securitygroup/securityrule.go:137]: This ensures the rule will be deleted when the service is deleted.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request closed without merging**